src/restconfig/src/main/java/org/geoserver/rest/resources/ResourceResource.java

```diff
@@ -229,10 +230,27 @@ private String href(String path, boolean isDir) {
         return href;
     }
 
-    public String formatHtmlLink(String link) {
+    private static String formatHtmlLink(String link) {
         return link.replaceAll("&", "&amp;");
     }
 
+    private static String formatPathURL(String input) {
```

Comment:
@NielsCharlier you are quite right. ```URLEncode.encode``` is intended for encoding form data in which spaces should be encoded as "+". In URL paths, spaces should be encoded as "%20". I am not sure which is the best practice code to use in GeoServer.